### PR TITLE
Use remote file dialog for database and server properties dialogs (#24390)

### PR DESCRIPTION
**Pull request by corivera**
Porting change to release branch.

Related issues: https://github.com/microsoft/azuredatastudio/issues/24313 and https://github.com/microsoft/azuredatastudio/issues/24318

### CODE REVIEW

File: src/sql/azdata.proposed.d.ts [hunk 1]
```diff
@@ -2068,9 +2068,14 @@ declare module 'azdata' {
...
 		 * @param fileFilters The filters used to limit which files are displayed in the file browser
+		 * @param showFoldersOnly Optional argument to specify whether the browser should only show folders
 		 * @returns The path of the file chosen from the dialog, and undefined if the dialog is closed without selecting anything.
 		 */
-		export function openServerFileBrowserDialog(connectionUri: string, targetPath: string, fileFilters: FileFilters[]): Thenable<string | undefined>;
+		export function openServerFileBrowserDialog(connectionUri: string, targetPath: string, fileFilters: FileFilters[], showFoldersOnly?: boolean): Thenable<string | undefined>;
+	}
+
+	export interface FileBrowserProvider extends DataProvider {
+		openFileBrowser(ownerUri: string, expandPath: string, fileFilters: string[], changeFilter: boolean, showFoldersOnly?: boolean): Thenable<boolean>;
```
Comment (Charles-Gagnon) [inline src/sql/azdata.proposed.d.ts:2078]: Did this never get the doc comment added to it? Please follow up with doing that in main....(it doesn't need to be ported over here)

**Review by Charles-Gagnon — APPROVED** [review]
It's usually best to not auto-close issues for the release branch, instead manually verifying and closing them after the next build to ensure that everything looks good.

File: src/sql/azdata.proposed.d.ts (hunk 1 above)
Comment (corivera) [reply src/sql/azdata.proposed.d.ts:2078]: Woops, merged the PR after I got home and forgot about that. I'll update it in main

Comment (corivera) [reply src/sql/azdata.proposed.d.ts:2078]: @Charles-Gagnon https://github.com/microsoft/azuredatastudio/pull/24413

**Review by kisantia — APPROVED** [review]

**Pull request merged**